### DISCUSSION TITLE
about the montgomery and edward curve public key validation

### DISCUSSION THREAD

**Discussion by steed717** [jedisct1/libsodium#1365] [Q&A]
Hi there,
I have 2 questions here:
1.  Does Libsodium have the Mont448 and Ed448 support now?
2.  In the FIPS NIST SP 800-186,  Appendix D 1.2 and D 1.23  ask for the public key validation.  Does Libsodium support it? If yes, via what API  functions?

Thanks,

**Comment by soatok** [comment]
1. No. See https://github.com/jedisct1/libsodium/tree/master/src/libsodium/crypto_scalarmult which only exposes the 25519 algorithms.

**Comment by jedisct1** [comment]
Public key validation is usually not required with Edwards25519, but if you really need to do it, use [`crypto_core_ed25519_is_valid_point`](https://libsodium.gitbook.io/doc/advanced/point-arithmetic#point-validation)
